### PR TITLE
K8s: Add output media type context helpers

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This adds request helpers that allow us to parse and stash the requested output format in the context.

**Why do we need this feature?**

To be able to identify requests for [Table output format](https://kubernetes.io/docs/reference/using-api/api-concepts/#receiving-resources-as-tables)

**Who is this feature for?**

@ryantxu 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/grafana-apiserver/endpoints/request/request.go [hunk 1]
```diff
@@ -24,3 +27,20 @@ func ParseOrgID(ns string) (int64, bool) {
 
 	return int64(orgID), true
 }
+
+type outputMediaType int
+
+const outputMediaKey outputMediaType = iota
```
Comment (ryantxu) [inline pkg/services/grafana-apiserver/endpoints/request/request.go:33]:
```suggestion
type outputMediaType struct{}
```

likely does not matter, but this is how we do context keys elsewhere

File: pkg/services/grafana-apiserver/endpoints/request/request.go [hunk 2]
```diff
@@ -24,3 +27,20 @@ func ParseOrgID(ns string) (int64, bool) {
...
+
+func WithOutputMediaType(ctx context.Context, req *http.Request, scope *handlers.RequestScope) context.Context {
+	outputMedia, _, err := negotiation.NegotiateOutputMediaType(req, scope.Serializer, scope)
```
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/services/grafana-apiserver/endpoints/request/request.go]: good question. all of the args would need to be stashed in context. i'm not sure what is better performance wise, but it seems like something we could optimize later if it becomes an issue. thoughts?

File: pkg/services/grafana-apiserver/endpoints/request/request.go (hunk 1 above)
Comment (toddtreece) [reply pkg/services/grafana-apiserver/endpoints/request/request.go]: 👍 switched

**Comment by toddtreece** [comment]
@ryantxu moving this back to draft. we'll most likely need access to request scope to use `negotiation.NegotiateOutputMediaType`. it's [created here](https://github.com/kubernetes/kubernetes/blob/0851995a61cb83645734183cb49250e0fc3b14a4/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L654) but there's no clear way to access it.

to me this seems like something we can punt on for now since it's a performance optimization and there are more basic building blocks that need work. thoughts?

File: pkg/services/grafana-apiserver/endpoints/request/request_test.go [hunk 3]
```diff
@@ -126,3 +136,105 @@ func TestParseNamespace(t *testing.T) {
...
+				Version: "v1",
+				Kind:    "Table",
+			},
+			ok: true,
+		},
+		{
+			name: "application/json",
+			req: &http.Request{
+				Header: http.Header{
+					"Accept": []string{"application/bson"},
```
Comment (mildwonkey) [inline pkg/services/grafana-apiserver/endpoints/request/request_test.go:167]: Should this be json? (if not maybe update the test name or add a comment to clarify it's supposed to mismatch)

Comment (toddtreece) [reply pkg/services/grafana-apiserver/endpoints/request/request_test.go:167]: good catch

**Comment by toddtreece** [comment]
@ryantxu i reached out to [#sig-api-machinery](https://kubernetes.slack.com/archives/C0EG7JC6T/p1697221280903029) in slack to see if there's a better way to do this

**Comment by toddtreece** [comment]
closing this for now

**Pull request closed without merging**